clients/admin-ui/src/features/custom-fields/CustomFieldForm.tsx

```diff
@@ -104,13 +104,25 @@ const CustomFieldForm = ({
       messageApi.error(getErrorMessage(result.error));
       return;
     }
-    // navigate to main page after success toast is shown
-    messageApi.success("Custom field deleted successfully", undefined, () => {
-      router.push(CUSTOM_FIELDS_ROUTE);
-    });
+    messageApi.success("Custom field deleted successfully");
+    router.push(CUSTOM_FIELDS_ROUTE);
   };
 
-  const [deleteModalIsOpen, setDeleteModalIsOpen] = useState(false);
+  const confirmDelete = () => {
+    modalApi.confirm({
+      title: "Delete custom field?",
+      content: (
+        <Typography.Paragraph>
+          Are you sure you want to delete <strong>{initialField?.name}</strong>{" "}
+          This action cannot be undone.
+        </Typography.Paragraph>
+      ),
+      onOk: handleDelete,
+      okType: "primary",
```

Comment:
should this be "danger"?
```suggestion
      okType: "danger",
```

Reply:
We don't tend to use danger buttons in confirmation modals, the modal itself is generally enough friction I think.  Maybe if the action was extremely destructive but I don't think this qualifies.